flocker/apiclient/_client.py

```diff
@@ -187,7 +219,7 @@ def list_datasets_configuration():
         Return the configured datasets, excluding any datasets that
         have been deleted.
 
-        :return: ``Deferred`` firing with iterable of ``Dataset``.
+        :return: ``Deferred`` firing with a ``DatasetsConfiguration``.
```

Comment:
:question: Do we care about backwards compatibility for the client (since this changes isn't compatible)? Although, `DatasetsConfiguration` could be made iterable to be backwards compatible with the documented interface.